### ISSUE TITLE
PIRANHA GUI - unclear error

### ISSUE THREAD

**Issue by ebujaki** [polio-nanopore/piranhaGUI#15]
Hi,
Our partners got the following issue, shown on the images, when they tried to run their DDNS data. (The trial sequencing run went well with some nice data to be analysed.)  Could you please advise what the problem might be? Thanks.

![Ghana_Screenshot 2024-02-06 144318](https://github.com/polio-nanopore/piranhaGUI/assets/126177398/fdd8fd97-fa22-4119-aef0-55988b6ecba3)
![Ghana_Screenshot 2024-02-06 144418](https://github.com/polio-nanopore/piranhaGUI/assets/126177398/740875a7-2175-47f1-abcc-82e1d80f9e65)

**Comment by aineniamh** [comment]
Hi @ebujaki, the issue is something has modified the supplementary data files and piranha is attempting to parse them, but cannot decode. Can you send me, even by email, the supplementary files that piranha is attempting to read? I'm not sure how this would have happened, but would be good to investigate.

**Comment by aineniamh** [comment]
I suspect the files were opened and modified and saved in some windows file viewer and now there are special hidden characters in the file.

**Comment by ebujaki** [comment]
Hi @aineniamh , I will check with the team. This would likely be a fasta for the tree?

**Comment by aineniamh** [comment]
Yeah, a fasta that was produced from the previous run and saved in the supplementary datadir- I wasn't aware piranhaGUI had this feature already!

**Comment by ebujaki** [comment]
Hi @aineniamh, I have not used a version of PIRANHA with this feature yet.

They have most likely not touched the file since the training when it was a new option. I don't know if a fasta was possibly added/edited or moved during training/demonstration on their laptop. This is optional, so presumably the feature could be temporarily disabled and the file wouldn't be read?

They suggested a call, which could well be the easiest way to sort it remotely.

**Comment by aineniamh** [comment]
It would be a feature toggled on with phylogenetics mode- it would be good to get @CorwinAnsley to comment, as I'm not aware of how piranhaGUI handles this (storing or accessing the file) if you don't think they have touched the file. 

If you disable the phylo mode (or don't supply a supplementary datadir, or specify a different one), then it will not have this error.

**Comment by ebujaki** [comment]
Okay, I will suggest they disable the module until I hear from Corey or get to check if redirecting/deleting the file also works. 

Could an example file with just 3 Sabin sequences be provided to show correct format?

**Comment by aineniamh** [comment]
Yeah, it's not clear what may have happened here- whether it's on the UI end or if the file has been modified in a windows application. 

Just a standard fasta format will work, but it will need to be saved as plain text (utf-8) rather than a rich text format that may contain hidden characters: 

```
>Sabin1-related
GGGTTAGGTCAGATGCTTGAAAGCATGATTGACAACACAGTCCGTGAAACGGTGGGGGCGGCAACGTCTAGAGACGCTCTCCCAAACACTGAAGCCAGTGGACCAGCACACTCCAAGGAAATTCCGGCACTCACCGCAGTGGAAACTGGGGCCACAAATCCACTAGTCCCTTCTGATACAGTGCAAACCAGACATGTTGTACAACATAGGTCAAGGTCAGAGTCTAGCATAGAGTCTTTCTTCGCGCGGGGTGCATGCGTGGCCATTATAACCGTGGATAACTCAGCTTCCACCAAGAATAAGGATAAGCTATTTACAGTGTGGAAGATCACTTATAAAGATACTGTCCAGTTACGGAGGAAATTGGAGTTCTTCACCTATTCTAGATTTGATATGGAATTTACCTTTGTGGTTACTGCAAATTTCACTGAGACTAACAATGGGCATGCCTTAAATCAAGTGTACCAAATTATGTACGTACCACCAGGCGCTCCAGTGCCCGAGAAATGGGACGACTACACATGGCAAACCTCATCAAATCCATCAATCTTTTACACCTACGGAACAGCTCCAGCCCGGATCTCGGTACCGTATGTTGGTATTTCGAACGCCTATTCACACTTTTACGACGGTTTTTCCAAAGTACCACTGAAGGACCAGTCGGCAGCACTAGGTGACTCCCTCTATGGTGCAGCATCTCTAAATGACTTCGGTATTTTGGCTGTTAGAGTAGTCAATGATCACAACCCGACCAAGGTCACCTCCAAAATCAGAGTGTATCTAAAACCCAAACACATCAGAGTCTGGTGCCCGCGTCCACCGAGGGCAGTGGCGTACTACGGCCCTGGAGTGGATTACAAGGATGGTACGCTTACACCCCTCTCCACCAAGGATCTGACCACATAT
>Sabin2-related
GGAATTGGTGACATGATTGAGGGGGCCGTTGAAGGGATTACTAAAAATGCATTGGTTCCCCCGACTTCCACCAATAGCCTGCCTGACACAAAGCCGAGCGGTCCAGCCCACTCCAAGGAGATACCTGCATTGACAGCCGTGGAGACAGGGGCTACCAATCCGTTGGTGCCTTCGGACACCGTGCAAACGCGCCATGTCATCCAGAGACGAACGCGATCAGAGTCCACGGTTGAGTCATTCTTTGCAAGAGGGGCTTGCGTGGCTATCATTGAGGTGGACAATGATGCACCGACAAAGCGCGCCAGCAGATTGTTTTCGGTTTGGAAAATAACTTACAAAGATACTGTTCAACTGAGACGCAAACTGGAATTTTTCACATATTCGAGATTTGACATGGAGTTCACTTTTGTGGTCACCTCAAACTACATTGATGCAAATAACGGACATGCATTGAACCAAGTTTATCAGATAATGTATATACCACCCGGAGCACCTATCCCTGGTAAATGGAATGACTATACGTGGCAGACGTCCTCTAACCCGTCGGTGTTTTACACCTATGGGGCGCCCCCAGCAAGAATATCAGTGCCCTACGTGGGAATTGCTAATGCGTATTCCCACTTTTATGATGGGTTTGCAAAAGTACCACTAGCGGGTCAAGCCTCAACTGAAGGCGATTCGTTGTACGGTGCTGCCTCACTGAATGATTTTGGATCACTGGCTGTTCGCGTGGTAAATGATCACAACCCCACGCGGCTCACCTCCAAGATCAGAGTGTACATGAAGCCAAAGCATGTCAGAGTCTGGTGCCCACGACCTCCACGAGCAGTCCCATACTTCGGACCAGGTGTTGATTATAAAGATGGGCTCACCCCACTACCAGAAAAGGGATTAACGACTTAT
>Sabin3-related
GGTATTGAAGATTTGACTTCTGAAGTTGCACAGGGCGCCCTAACTTTGTCACTCCCGAAGCAACAGGATAGCTTACCTGATACTAAGGCCAGTGGCCCGGCGCATTCCAAGGAGGTACCTGCACTCACTGCAGTCGAGACTGGAGCCACCAATCCTCTGGCACCATCCGACACAGTTCAAACGCGCCACGTAGTCCAACGACGCAGCAGGTCAGAGTCCACAATAGAATCATTCTTCGCACGCGGGGCGTGCGTCGCTATTATTGAGGTGGACAATGAACAACCAACCACCCGGGCACAGAAACTATTTGCCATGTGGCGCATTACATACAAAGATACAGTGCAGTTGCGCCGTAAGTTGGAGTTTTTCACATACTCTCGTTTTGACATGGAATTCACCTTCGTGGTAACCGCCAACTTCACCAACGCTAATAATGGGCATGCACTCAACCAGGTGTACCAGATAATGTACATCCCCCCAGGGGCACCCACACCAAAGTCATGGGACGACTACACTTGGCAAACATCTTCCAACCCGTCCATATTTTACACCTATGGGGCTGCCCCGGCGCGAATCTCAGTGCCATACGTGGGGTTAGCCAATGCTTACTCGCACTTTTACGACGGCTTCGCCAAGGTGCCATTGAAGACAGATGCCAATGACCAGATTGGTGATTCCTTGTACAGCGCCATGACAGTTGATGACTTTGGTGTATTGGCAGTTCGTGTTGTCAATGATCACAACCCCACTAAAGTAACCTCCAAAGTCCGCATTTACATGAAACCCAAACACGTACGTGTCTGGTGCCCTAGACCGCCGCGCGCGGTACCTTATTATGGACCAGGGGTGGACTATAGGAACAACTTGGACCCCTTATCTGAGAAAGGTTTGACCACATAT
```

**Comment by ebujaki** [comment]
Thank you, very useful tip!

**Comment by AShaw1802** [comment]
We've just been checking this again, and there's an issue with the entry of the folder address for the phylogenetics module in PiranhaGUI- once you select one, there's no way to clear it, even if you restart.  Unless you have a folder with the correctly formatted data to hand you can no longer use the module, as even an empty folder causes Piranha to crash (as far as I can tell). If PiranhaGUI could clear the address and/or phlogenetics without the folder I think this might solve the problem. 

Just confirmed that if I drop the example data from above into the folder as a fasta file then Piranha works as expected.

**Comment by CorwinAnsley** [comment]
Hi, sorry for the late response. I haven't been able to recreate that error so it does seem like the files were modified somehow already. PiranhaGUI shouldn't be modifying them in any case, it's mounting the provided directory in the docker container like it does the run folder. I'll try to implement some kind of check to warn the user if the supplementary data file has unicode characters that will cause errors.

As for the inability to clear the address for the phylo directory that's definitely an oversight, I've implemented a button to do that on the dev branch for the next release

**Comment by ebujaki** [comment]
Hi,
Okay, thanks for checking. We will look out for the update in the next release and replace the current file for one in the correct format.
